TYPO3.Flow/Classes/TYPO3/Flow/Package/PackageFactory.php

```diff
@@ -41,15 +43,31 @@ public function create($packagesBasePath, $packagePath, $packageKey, $composerNa
         }
 
         $packageClassName = Package::class;
+        $packageClassExists = true;
         if (!empty($packageClassInformation)) {
             $packageClassName = $packageClassInformation['className'];
             $packageClassPath = Files::concatenatePaths(array($absolutePackagePath, $packageClassInformation['pathAndFilename']));
-            require_once($packageClassPath);
+            $packageClassExists = @include_once($packageClassPath);
+
+            if ($packageClassExists === false) {
+                throw new Exception\PackageClassMissingException(
+                    sprintf(
+                        'The package class "%s" (expected in file "%s" of package "%s") could not be loaded.',
+                        $packageClassName,
+                        $packageClassPath,
+                        $$packageKey
+                    ), 1461911652);
```

Comment:
this check is to make sure that it works when `$packageClassInformation` is provided directly to the function right? because this shouldn't be possible when it comes from `detectFlowPackageFilePath`